rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -79,36 +77,53 @@ void Recorder::record(const RecordOptions & record_options)
   subscriptions_.clear();
 }
 
-void Recorder::topics_discovery(
-  std::chrono::milliseconds topic_polling_interval,
-  const std::vector<std::string> & requested_topics,
-  bool include_hidden_topics)
+void Recorder::topics_discovery(const RecordOptions & record_options)
 {
   while (rclcpp::ok()) {
     auto topics_to_subscribe =
-      get_requested_or_available_topics(requested_topics, include_hidden_topics);
+      get_requested_or_available_topics(record_options);
     for (const auto & topic_and_type : topics_to_subscribe) {
       warn_if_new_qos_for_subscribed_topic(topic_and_type.first);
     }
     auto missing_topics = get_missing_topics(topics_to_subscribe);
     subscribe_topics(missing_topics);
 
-    if (!requested_topics.empty() && subscriptions_.size() == requested_topics.size()) {
+    if (!record_options.topics.empty() && subscriptions_.size() == record_options.topics.size()) {
       ROSBAG2_TRANSPORT_LOG_INFO("All requested topics are subscribed. Stopping discovery...");
       return;
     }
-    std::this_thread::sleep_for(topic_polling_interval);
+    std::this_thread::sleep_for(record_options.topic_polling_interval);
   }
 }
 
 std::unordered_map<std::string, std::string>
-Recorder::get_requested_or_available_topics(
-  const std::vector<std::string> & requested_topics,
-  bool include_hidden_topics)
+Recorder::get_requested_or_available_topics(const RecordOptions & record_options)
 {
-  return requested_topics.empty() ?
-         node_->get_all_topics_with_types(include_hidden_topics) :
-         node_->get_topics_with_types(requested_topics);
+  auto tt = record_options.topics.empty() ?
```

Comment:
Please avoid one or two character variable names that aren't loop variables (and even then it doesn't hurt to name those better) - even in relatively short functions these makes for difficult reading. Maybe `desired_topics`?